flash/data/process.py

```diff
@@ -0,0 +1,188 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, Optional, Sequence, Union
+
+import torch
+from pytorch_lightning.trainer.states import RunningStage
+from torch.nn import Module
+from torch.utils.data._utils.collate import default_collate
+
+from flash.data.batch import default_uncollate
+from flash.data.utils import convert_to_modules
+
+
+class Properties:
+
+    _running_stage = None
```

Comment:
interesting stage sharing :+1:

Reply:
Yeah, might be worth to bring in Lightning at some point.